.htaccess

```diff
@@ -1,240 +1,561 @@
+	BrowserMatchNoCase "Widow(.*)" BAD_BOT
+	BrowserMatchNoCase "WWWOFFLE(.*)" BAD_BOT
+	BrowserMatchNoCase "Xaldon WebSpider(.*)" BAD_BOT
+	BrowserMatchNoCase "Zeus(.*)" BAD_BOT
+	BrowserMatchNoCase "zgrab(.*)" BAD_BOT
+	BrowserMatchNoCase "ZoominfoBot(.*)" BAD_BOT
+
+###########################################
+## Deny access (Default)
+## Comment the lines if you use the alternative method
+
+	Order allow,deny
+	Deny from env=BAD_BOT
+
+###########################################
+## Redirect to external location (Alternative)
+## Apache Module mod_rewrite must be enabled
 
-        #AuthName "Cron auth"
-        #AuthUserFile ../.htpasswd
-        #AuthType basic
-        #Require valid-user
+	#RewriteCond %{env:BAD_BOT} !^$
+	#RewriteRule ^(.*)$ https://www.example.com/ [L,R=301]
+
+</IfModule>
+
+<IfModule mod_rewrite.c>
 
 ############################################
+## The ultimate hotlink protection
+## Before use set the YOUR_DOMAIN_NAME value in the last condition 
+## https://perishablepress.com/creating-the-ultimate-htaccess-anti-hotlinking-strategy/
+
+	RewriteEngine on
+	RewriteCond %{HTTP_REFERER} !^$
+	RewriteCond %{REQUEST_FILENAME} -f
+	RewriteCond %{REQUEST_FILENAME} \.(gif|jpe?g?|png)$ [NC]
+	RewriteCond %{HTTP_REFERER} !^https?://([^.]+\.)?YOUR_DOMAIN_NAME\. [NC]
+	
+###########################################
+## Always response 403 Forbidden status code (Default)
+## Comment the line if you use the alternative method
+
+	#RewriteRule \.(gif|jpe?g?|png)$ - [NC,L,R=403]
+
+###########################################
+## Use a generic image for all requests (Alternative)
+## Before use change the DOMAIN.TLD and HOTLINK values
 
-        Order allow,deny
-        Deny from all
+	#RewriteRule \.(gif|jpe?g?|png)$ https://DOMAIN.TLD/HOTLINK.jpg [NC,L,R=301]
 
-    </Files>
+</IfModule>
+
+<IfModule mod_alias.c>
+
+###########################################
+## Redirect access to catalog feeds pages
+
+	#RedirectMatch 301 /rss/catalog/catalog/review(.*)$	https://www.example.com
+	#RedirectMatch 301 /rss/catalog/notifystock(.*)$	https://www.example.com
+	#RedirectMatch 301 /rss/catalog/order/new(.*)$		https://www.example.com
+
+###########################################
+## Redirect access to Categories and Products pages based on IDs
+
+	#RedirectMatch 301 /catalog/category/view/id(.*)$	https://www.example.com
+	#RedirectMatch 301 /catalog/product/view/id(.*)$	https://www.example.com
+
+###########################################
+## Redirect access to Categories and Products sitemap pages 
+	
+	#RedirectMatch 301 /catalog/seo_sitemap/category(.*)$	https://www.example.com
+	#RedirectMatch 301 /catalog/seo_sitemap/product(.*)$	https://www.example.com
+
+###########################################
+## Redirect access to Advanced Search and Popular Search Terms pages
+
+	#RedirectMatch 301 /catalogsearch/advanced(.*)$		https://www.example.com
+	#RedirectMatch 301 /catalogsearch/term/popular(.*)$	https://www.example.com
```

Comment:
people know how to make a 301 redirect and we shouldn't have this basic sections here, it's really easy to find the proper documentation in the right places